src/olympia/shelves/forms.py

```diff
@@ -19,27 +19,29 @@ def clean(self):
         criteria = data['criteria']
         baseUrl = settings.INTERNAL_SITE_URL
 
-        if data['shelf_type'] in ('extension', 'search', 'theme'):
+        if data['shelf_type'] in ('extension', 'recommended', 'search',
+                                  'theme'):
             api = drf_reverse('v4:addon-search')
+            url = baseUrl + api + criteria
         elif data['shelf_type'] == 'categories':
             api = drf_reverse('v4:category-list')
```

Comment:
I didn't look too closely at the actual api endpoints defined before - the categories shelves don't access the category list (check the network panel on AMO frontend) - it's an addons/search query.    They're _all_ addons/search queries apart from the collection shelves.

Reply:
I'll update this.